storage/series/reader.go

```diff
@@ -226,7 +234,9 @@ func (r Reader) fetchBlocksWithBlocksMapAndBuffer(
 					res = append(res, r)
 				}
 				if stream != nil {
-					r := block.NewFetchBlockResult(start, []xio.SegmentReader{stream}, nil)
+					end := start.Add(blockDuration)
```

Comment:
As per previous comment, instead of passing the block duration / block size here,  let's just make the underlying blocks return an `xio.BlockReader`.